### PR TITLE
Add filter

### PR DESCRIPTION
**Pull request by pascutto**
/cc @liautaud

### CODE REVIEW

File: src/index.ml [hunk 1]
```diff
@@ -547,8 +549,9 @@ struct
       append_entry_fanout fan_out log.(log_i) dst_io
     done
 
-  (* Merge [log] with [t] into [dst_io]. [log] must be sorted by key hashes. *)
-  let merge_with log (index : index) dst_io =
+  (* Merge [log] with [index] into [dst_io], ignoring keys that do not satisfy
+     [predicate]. [log] must be sorted by key hashes. *)
```
Comment (CraigFe) [inline src/index.ml]: `[filter]`

File: src/index.mli [hunk 2]
```diff
@@ -169,6 +169,10 @@ module type S = sig
   (** [replace t k v] binds [k] to [v] in [t], replacing any existing binding of
       [k]. *)
 
+  val filter : t -> (key * value -> bool) -> unit
+  (** [filter t p] triggers a blocking merge where only bindings that satisfy
+      [p] are kept. *)
```
Comment (CraigFe) [inline src/index.mli]: This is an odd description to be user-facing: can we drop the reference to 'merge'?

**Comment by pascutto** [comment]
Already fixed just one minute before your review ;)

File: src/index.mli [hunk 3]
```diff
@@ -169,6 +169,10 @@ module type S = sig
   (** [replace t k v] binds [k] to [v] in [t], replacing any existing binding of
       [k]. *)
 
+  val filter : t -> (key * value -> bool) -> unit
+  (** [filter t p] removes all the bindings (k, v) that do not satisfy [p]. This
+      operation is costly and blocking. *)
```
[the remark this answers is not in the document]
Comment (samoht) [reply src/index.mli]: Actually as merge is not in the API that’s probably not very useful :-)

**Pull request merged**